File: codewar/6-kyu/split_string.py
# solved
def solution(string: str) -> list[str]:
    if len(string) == 0:
        return []

    res = []
    for char in range(0, len(string), 2):
        res.append(string[char : char + 2])

    if len(res[len(res) - 1]) == 1:
        res[len(res) - 1] += "_"

    return res


a_string = "abcdeb"
n = 2
split_strings = [a_string[index : index + n] for index in range(0, len(a_string), n)]

split_strings[len(split_strings) - 1] += "_"
# print(split_strings[len(split_strings) - 1])
# print(split_strings)
# print(solution(""))

# ------------------------------ other solutoin ------------------------------ #
# with regex
import re


def withRegex(string: str) -> "list[str]":
    return re.findall(".{2}", string + "_")


b = "abcd"
print(withRegex(b))

File: codewar/6-kyu/splitString/splitString.go
package main

import (
	"fmt"
)

func Solution(str string) []string {
	var res []string

	if len(str)%2 != 0 {
		str += "_"
	}

	for i := 0; i < len(str); i += 2 {
		res = append(res, string(str[i:i+2]))
	}
	fmt.Println(res)
	return res
}

func main() {
	a := "daniel"
	fmt.Println(len(a), a[4:6])
	// fmt.Println(Solution("danie"))
}
